jwst/lib/dispaxis.py

```diff
@@ -0,0 +1,152 @@
+                             pupil="ANY"):
+    """Get the dispersion direction.
+
+    Parameters
+    ----------
+    exposure_type : str
+        The exposure type.
+
+    grating : str
+        The name of the optic in the grating wheel.
+
+    filter : str
+        The name of the optic in the filter wheel.
+
+    pupil : str
+        The name of the optic in the pupil wheel.
+
+    Returns
+    -------
+    int : The dispersion direction
+
+-       0  the dispersion direction is not meaningful or not defined
+-       1  the dispersion direction is horizontal ("sky" coordinates)
+-       2  the dispersion direction is vertical ("sky" coordinates)
+    """
+
+    exposure_type = exposure_type.upper()
+    grating = grating.upper()
+    filter = filter.upper()
+    pupil = pupil.upper()
+
+    # The keys of the `by_exp_type` dictionary are values of exposure type.
+    # If the dispersion direction is uniquely determined by the exposure
+    # type, the associated value of by_exp_type[exposure_type] will be an
+    # integer, the dispersion direction.  If one or more parameters other
+    # than the exposure type are needed, the value of
+    # by_exp_type[exposure_type] will be a tuple of several strings; use
+    # that tuple as the key for dictionary `second_pass`, and the value of
+    # that will be the dispersion direction.
+    by_exp_type = {
+        # FGS science
+        "FGS_DARK": 0,
+        "FGS_FOCUS": 0,
+        "FGS_IMAGE": 0,
+        "FGS_INTFLAT": 0,
+        "FGS_SKYFLAT": 0,
+        # FGS guide star
+        "FGS_ACQ1": 0,
+        "FGS_ACQ2": 0,
+        "FGS_FINEGUIDE": 0,
+        "FGS_ID-IMAGE": 0,
+        "FGS_ID-STACK": 0,
+        "FGS_TRACK": 0,
+        # MIRI
+        "MIR_4QPM": 0,
+        "MIR_CORONCAL": 0,
+        "MIR_DARKALL": 0,
+        "MIR_DARKIMG": 0,
+        "MIR_DARKMRS": 0,
+        "MIR_FLATALL": 0,
+        "MIR_FLATIMAGE": 0,
+        "MIR_FLATIMAGE-EXT": 0,
+        "MIR_FLATMRS": 2,
+        "MIR_FLATMRS-EXT": 2,
+        "MIR_IMAGE": 0,
+        "MIR_LRS-FIXEDSLIT": 2,
+        "MIR_LRS-SLITLESS": 2,
+        "MIR_LYOT": 0,
+        "MIR_MRS": 2,
+        "MIR_TACQ": 0,
+        # NIRISS
+        "NIS_AMI": 0,
+        "NIS_DARK": 0,
+        "NIS_EXTCAL": 0,
+        "NIS_FOCUS": 0,
+        "NIS_IMAGE": 0,
+        "NIS_LAMP": 0,
+        "NIS_SOSS": 1,
+        "NIS_TACQ": 0,
+        "NIS_TACONFIRM": 0,
+        "NIS_WFSS": (exposure_type, "ANY", filter, "ANY"),
+        # NIRCam
+        "NRC_CORON": 0,
+        "NRC_DARK": 0,
+        "NRC_FLAT": 0,
+        "NRC_FOCUS": 0,
+        "NRC_GRISM": (exposure_type, "ANY", "ANY", pupil),
+        "NRC_IMAGE": 0,
+        "NRC_WFSS": (exposure_type, "ANY", "ANY", pupil),
+        "NRC_LED": 0,
+        "NRC_WFSC": 0,
+        "NRC_TACONFIRM": 0,
+        "NRC_TACQ": 0,
+        "NRC_TSGRISM": (exposure_type, "ANY", "ANY", pupil),
+        "NRC_TSIMAGE": 0,
+        # NIRSpec
+        "NRS_AUTOFLAT": 0,
+        "NRS_AUTOWAVE": 1,
```

Comment:
This is currently not processed by assign_wcs. Should it?

Reply:
I doubt that this should be processed by assign_wcs.